src/prefect/blocks/abstract.py

```diff
@@ -302,3 +302,61 @@ async def upload_from_folder(
         Returns:
             The path that the folder was uploaded to.
         """
+
+
+class SecretBlock(Block, ABC):
+    """
+    Block that represents a resource that can store and retrieve secrets.
+    """
+
+    @property
+    def logger(self) -> Logger:
+        """
+        Returns a logger based on whether the SecretBlock
+        is called from within a flow or task run context.
+        If a run context is present, the logger property returns a run logger.
+        Else, it returns a default logger labeled with the class's name.
+        """
+        try:
+            return get_run_logger()
+        except MissingContextError:
+            return get_logger(self.__class__.__name__)
+
+    @abstractmethod
+    async def read_secret(self):
```

Comment:
I think this should have a return type and a description of what it is expect to return.